conf/app.ini

```diff
@@ -46,6 +46,10 @@ DOMAIN = localhost
 ROOT_URL = %(PROTOCOL)s://%(DOMAIN)s:%(HTTP_PORT)s/
 HTTP_ADDR =
 HTTP_PORT = 3000
+; Local (DMZ) URL for gogs workers (such as ssh update) accessing web service.
+; In most cases you may leave it empty. Defaults: http://localhost:HTTP_PORT/
+; You may need to set it only if ssh server node is not the same as http node.
+LOCAL_ROOT_URL =
```

Comment:
It just a INI syntax for recursion values, the value you get for `LOCAL_ROOT_URL` is changing with the change of `HTTP_PORT`. 

It does not show in `custom/conf/app.ini` if user does not touch it.